code/game/objects/items/stacks/stack.dm

```diff
@@ -89,6 +93,14 @@
 		icon_state = "[initial(icon_state)]_2"
 	else
 		icon_state = "[initial(icon_state)]_3"
+
+	if(GetComponent(overlay))
+		if(src.GetComponent(/datum/component/decal/shimmer))
+			S.TakeComponent(src)
+			src.AddComponent(/datum/component/decal/shimmer)
+		if(src.GetComponent(/datum/component/decal/blood))
+			B.TakeComponent(src)
+			src.AddComponent(/datum/component/decal/blood)
 	..()
```

Comment:
There is surely a better way of doing this? This syntax looks kind of odd.

Reply:
Almost certainly there is, this is trying to fix that stacked items overlay thing, but I haven't had any luck with a fix so far, so this'll likely disappear once I get that worked out.